### PR TITLE
Added check to support RelStorage

### PR DESCRIPTION
**Pull request by tflorac**
Update to support history-free RelStorage support.

### CODE REVIEW

File: src/zodbupdate/update.py [hunk 1]
```diff
@@ -146,7 +148,8 @@ def records(self):
                 if next is None:
                     break
         elif (IStorageIteration.providedBy(storage) and
-              not storage.supportsUndo()):
+              (not IStorageUndoable.providedBy(self.storage) or
+               not self.storage.supportsUndo())):
```
Comment (mgedmin) [inline src/zodbupdate/update.py]: Why `self.storage` when everything else is checked and done on `storage`?

(AFAICT BlobStorage copies the interfaces and delegates the supportsUndo implementation so it shouldn't matter in practice.)

Comment (tflorac) [reply src/zodbupdate/update.py]: Oops!
My mistake, sorry!!
I pushed a corrected version...  ;)

File: src/zodbupdate/update.py [hunk 2]
```diff
@@ -12,19 +12,21 @@
...
 import six
-import io
-import logging
 import transaction
 import zodbupdate.serialize
 import zodbupdate.utils
+from ZODB.FileStorage import FileStorage
+from ZODB.blob import BlobStorage
+from ZODB.interfaces import IStorageCurrentRecordIteration, IStorageIteration, IStorageUndoable
+
```
Comment (icemac) [inline src/zodbupdate/update.py]: Are these changes in the ordering of the imports needed for the fix, if not please revert them.

**Review by icemac — CHANGES_REQUESTED** [review]
The change seems reasonable. I'd request a change log entry and undoing the reordering of the imports.

File: src/zodbupdate/update.py (hunk 2 above)
Comment (tflorac) [reply src/zodbupdate/update.py]: Of course they are not!  :)
I submitted a new patch...

**Comment by icemac** [comment]
@tflorac Thank you for this PR, according to https://github.com/zopefoundation/zope.sqlalchemy/pull/39#issuecomment-563288848 signing a contributor agreement is required so this PR can be merged. See https://www.zope.org/developer/becoming-a-committer.html.

**Comment by mgedmin** [comment]
> The change seems reasonable. I'd request a change log entry and undoing the reordering of the imports.

Why?  The import reordering was small and it fixed a previous non-PEP-8-compliant mess.

**Comment by tflorac** [comment]
> @tflorac Thank you for this PR, according to [zopefoundation/zope.sqlalchemy#39 (comment)](https://github.com/zopefoundation/zope.sqlalchemy/pull/39#issuecomment-563288848) signing a contributor agreement is required so this PR can be merged. See https://www.zope.org/developer/becoming-a-committer.html.

I filled and sent the contributor agreement.
Don't know how long it is to take effect...

**Comment by tflorac** [comment]
> > The change seems reasonable. I'd request a change log entry and undoing the reordering of the imports.
> 
> Why? The import reordering was small and it fixed a previous non-PEP-8-compliant mess.

In fact... I was planning to refactor all package imports in PEP8 format...!
Is there a general rule against that?

**Comment by mgedmin** [comment]
> In fact... I was planning to refactor all package imports in PEP8 format...!
> Is there a general rule against that?

Not as far as I know!  There's the usual advice to avoid mixing refactorings with other changes in one PR, to keep diffs readable, but I found the original diff perfectly readable as it was sufficiently small.

**Comment by jugmac00** [comment]
In an ideal world there would be both local tools (e.g. pre-commit) and server-side linters to enforce one common standard for all Zope repositories, so there would be no need to discuss such vanities.

Also, there should be **written** developer guidelines, with detailed information how one can contribute and what to take care of. Especially the fact that new contributors need to sign an agreement should be well known in advance. I think this could be really frustrating if one spends quite some time for a pr and then cannot merge, maybe because the employer does not allow its employees to sign such an agreement.

We have some information at https://www.zope.org/developer/index.html - but a) its outdated and b) those information should be available directly from the repositories readme files (via link).

**Comment by tflorac** [comment]
> Not as far as I know! There's the usual advice to avoid mixing refactorings with other changes in one PR, to keep diffs readable, but I found the original diff perfectly readable as it was sufficiently small.

OK, so I merged the first PR and I will provide another one with refactored imports...  ;)

**Pull request merged**